packages/next/src/server/dev/hot-reloader-turbopack.ts

```diff
@@ -612,6 +612,19 @@ export async function createHotReloaderTurbopack(
         )
       }
 
+      const existingRoutes = [
+        ...currentEntrypoints.app.keys(),
+        ...currentEntrypoints.page.keys(),
+      ]
+      const newRoutes = [...entrypoints.routes.keys()]
+
+      const addedRoutes = newRoutes.filter(
+        (route) => !existingRoutes.includes(route)
+      )
+      const removedRoutes = existingRoutes.filter(
+        (route) => !newRoutes.includes(route)
+      )
```

Comment:
Using `Array.includes` is not really efficient here (with large number of routes), but you could make it efficient really easily as the original lists are already Maps.

```suggestion
      const addedRoutes = newRoutes.filter(
        (route) => !currentEntrypoints.app.has(route) && !currentEntrypoints.page.has(route)
      )
      const removedRoutes = existingRoutes.filter(
        (route) => !entrypoints.routes.has(route)
      )
```

Reply:
Applied in #83840